### DISCUSSION TITLE
API 관련된 이야기를 나눕니다.

### DISCUSSION THREAD

**Discussion by jgjgill** [prgrms-fe-devcourse/FEDC4_HONKOK_JunilHwang#19] [Q&A]
API 명세서 중 이해가 안가는 부분 or 참고할 만한 부분들에 대해 의견을 나눕니다.

편하게 작성해주세요.

**Comment by backward99** [comment]
어떤 방식을 사용해서 api를 숨기는게 좋을까요? vercel을 사용하는 방향으로 진행하게 될까요?

**Reply by jgjgill** [reply]
저한테는 api를 숨기는건 후순위였는데요..! 😇
기능 구현 관련 api를 먼저 하고 싶어서 다른 분들 의견이 궁금하네요.. 😂

**Comment by jgjgill** [comment]
react query 관련해서 공식 문서와 같이 읽으면 좋을 것 같은 자료 공유합니다..

- [react-query-tutorial](https://github.com/ssi02014/react-query-tutorial)
  - 한글로 되어 있는게 가장 큰 장점인 것 같네요.
- [Practical React Query](https://tkdodo.eu/blog/practical-react-query)
  - react query 메인테이너가 작성한 시리즈 글이어서 깊게 학습할 때 좋을 것 같네요.

**Reply by jgjgill** [reply]
쿼리 키 관련 글
[Effective React Query Keys](https://tkdodo.eu/blog/effective-react-query-keys)

**Comment by backward99** [comment]
글을 작성할 때 글 제목, 글 내용을 받아와서 사용하는 부분
```
const json = {
  "title": "제목",
  "body": "내용"
};

API.createPost(JSON.stringify(json), ...);
```
이 부분이 그냥 title에 해당 내용을 다 넣고 받아온 다음에는 JSON.parse(title)를 통해서 title.title, title.body 이런 식으로 사용하는 것 같다고 이해됩니다..

**Reply by jgjgill** [reply]
이해했습니다..! 참고해서 만들어볼게요!! 🙇‍♂️

**Comment by jgjgill** [comment]
<img width="761" alt="스크린샷 2023-09-18 오후 11 09 51" src="https://github.com/prgrms-fe-devcourse/FEDC4_HONKOK_JunilHwang/assets/79239852/094017fb-98a2-4c3e-a69c-88ec57df26d7">

해당 API는 `GET /channels/{channelName}`가 올바른 주소입니다.
